pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -247,4 +250,12 @@ public void onBecomeDroppedFromError(Message message, NotificationContext contex
       }
     }
   }
+
+  @Override
+  public ExecutorService getExecutorService(String resourceName, String fromState, String toState) {
```

Comment:
ideally lets add a seperate flag to create a threadpool per resource (table). that way we achieve more isolation

Reply:
yes we need to configure it appropriatley. Overall, I think we need 3 behaviours
1. default - use existing helix thread pool executor
2. thread pool based on state transition type (already added in the PR). I agree this is a more manageable conf
3. thread pool based on resource name. This will be required in extreme cases where tables are stepping on each other.  Having this capability will enable configuring it when required.